dateparser/date_parser.py

```diff
@@ -73,7 +75,11 @@ def _populate(cls, res, default):
             value = getattr(res, field)
             if value is not None:
                 repl[field] = value
-        new_date = new_date.replace(**repl)
+
+        # Fix day and populate date with result
+        repl_copy = repl.copy()
```

Comment:
Corrected `repl` interferes later on with correcting dates for past/future. Hence, we are using copy here.